src/test/java/bio/terra/service/dataset/DatasetConnectedTest.java

```diff
@@ -417,4 +435,320 @@ public void testSharedLockTableIngest() throws Exception {
+        assertEquals("Size of live table is still 3", 3, liveTableRowIds2.size());
+        assertFalse("Soft deleted row id is still not in live table", liveTableRowIds2.contains(softDeleteRowId));
+
+        // check that the size of the soft delete table has not changed
+        List<String> softDeleteRowIds2 = getRowIdsFromBQTable(summaryModel.getName(), internalSoftDeleteTableName);
+        assertEquals("Size of soft delete table is still 1", 1, softDeleteRowIds2.size());
+        assertTrue("Soft deleted row id is still in soft delete table", softDeleteRowIds2.contains(softDeleteRowId));
+
+        // delete the dataset and check that it succeeds
+        connectedOperations.deleteTestDataset(summaryModel.getId());
+
+        // try to fetch the dataset again and confirm nothing is returned
+        connectedOperations.getDatasetExpectError(summaryModel.getId(), HttpStatus.NOT_FOUND);
+    }
+
+    @Test
+    public void testConcurrentSoftDeletes() throws Exception {
+        // create a dataset and check that it succeeds
+        String resourcePathDatasetCreate = "snapshot-test-dataset.json";
+        DatasetRequestModel datasetRequest =
+            jsonLoader.loadObject(resourcePathDatasetCreate, DatasetRequestModel.class);
+        datasetRequest
+            .name(Names.randomizeName(datasetRequest.getName()))
+            .defaultProfileId(billingProfile.getId());
+        DatasetSummaryModel summaryModel = connectedOperations.createDataset(datasetRequest);
+
+        // load a CSV file that contains the table rows to load into the test bucket
+        String resourceFileName = "snapshot-test-dataset-data.csv";
+        String dirInCloud = "scratch/testConcurrentSoftDeletes/" + UUID.randomUUID().toString();
+        BlobInfo ingestTableBlob = BlobInfo
+            .newBuilder(testConfig.getIngestbucket(), dirInCloud + "/" + resourceFileName)
+            .build();
+        Storage storage = StorageOptions.getDefaultInstance().getService();
+        storage.create(ingestTableBlob,
+            IOUtils.toByteArray(getClass().getClassLoader().getResource(resourceFileName)));
+        String tableIngestInputFilePath =
+            "gs://" + testConfig.getIngestbucket() + "/" + dirInCloud + "/" + resourceFileName;
+
+        // ingest the table
+        String tableName = "thetable";
+        IngestRequestModel ingestRequest = new IngestRequestModel()
+            .table(tableName)
+            .format(IngestRequestModel.FormatEnum.CSV)
+            .csvSkipLeadingRows(1)
+            .path(tableIngestInputFilePath);
+        connectedOperations.ingestTableSuccess(summaryModel.getId(), ingestRequest);
+
+        // make sure the JSON file gets cleaned up on test teardown
+        connectedOperations.addScratchFile(dirInCloud + "/" + resourceFileName);
+
+        // load CSV file #1 that contains the table rows to soft delete into the test bucket
+        String softDeleteRowId1 = "8c52c63e-8d9f-4cfc-82d0-0f916b2404c1";
```

Comment:
Thought the same thing -- this isn't using the new ingest path, it's using the old one where we loaded the rows in (and therefor the row ids come from the test file).

Reply:
Right, the old path was more convenient for testing because I can just pull the rowid from the file. If we ever get rid of that though, then I could just add in a query to get the rowids from BQ before building the soft delete payload.